ui-v2/src/hooks/variables.ts

```diff
@@ -11,82 +11,91 @@ import {
+	variable: components["schemas"]["VariableUpdate"] & {
+		id: string;
 	},
+) => {
+	const { id, ...body } = variable;
+	return getQueryService().PATCH("/variables/{id}", {
+		params: { path: { id } },
+		body,
+	});
 };
 
-/**
- * Builds a query configuration for fetching filtered variables
- * @param options - Filter options for the variables query including pagination, sorting, and filtering criteria
- * @returns A query configuration object compatible with TanStack Query including:
- *  - queryKey: A unique key for caching the query results
- *  - queryFn: The async function to fetch the filtered variables
- *  - staleTime: How long the data should be considered fresh (1 second)
- *  - placeholderData: Uses previous data while loading new data
- */
-const buildVariablesQuery = (options: UseVariablesOptions) =>
-	queryOptions({
-		queryKey: variableKeys.filtered(options),
-		queryFn: async () => {
-			const response = await getQueryService().POST("/variables/filter", {
-				body: options,
-			});
-			return response.data;
-		},
-		staleTime: 1000,
-		placeholderData: keepPreviousData,
+const deleteVariableRequest = (id: string) => {
+	return getQueryService().DELETE("/variables/{id}", {
+		params: { path: { id } },
 	});
+};
+
+// ----- 🔑 Queries 🗄️
+// ----------------------------
 
 /**
- * Builds a query configuration for counting variables with optional filters
- * @param options - Optional filter options for the variables count query
- * @returns A query configuration object compatible with TanStack Query including:
- *  - queryKey: A unique key for caching the count results
- *  - queryFn: The async function to fetch the filtered count
- *  - staleTime: How long the data should be considered fresh (1 second)
- *  - placeholderData: Uses previous data while loading new data
+ * ```
+ *  🏗️ Variable queries construction 👷
+ *  all   =>   ['variables'] // key to match ['variables', ..., ...]
+ *  list  =>   ['variables', 'list'] // key to match ['variables', 'list', ..., ...]
+ *             ['variables', 'list', { ...withOptionsFoo }]
+ *             ['variables', 'list', { ...withOptionsBar }]
+ *             ['variables', 'list', { ...withOptionsBaz }]
+ *  count =>   ['variables', 'count'] // key to match ['variables', 'count', ..., ...]
+ *             ['variables', 'count', {}] // all count
+ *             ['variables', 'count', { ...withOptionsFoo }]
+ *             ['variables', 'count', { ...withOptionsBar }]
+ *             ['variables', 'count', { ...withOptionsBaz }]
+ * ```
  */
-const buildCountQuery = (options?: UseVariablesOptions) =>
-	queryOptions({
-		queryKey: variableKeys.filteredCount(options),
-		queryFn: async () => {
-			const body = options?.variables ? { variables: options.variables } : {};
-			const response = await getQueryService().POST("/variables/count", {
-				body,
-			});
-			return response.data;
-		},
-		staleTime: 1000,
-		placeholderData: keepPreviousData,
-	});
+const variableQueries = {
```

Comment:
It strikes me as odd that some of these functions return a query key while others return a full `queryOptions` object. The heterogenous return types make me think these should be grouped differently.

Reply:
Ya, might need to go back to the drawing board on this one.
It is a bit strange that the objects are inconsistent, but I _do_ agree with the blog on how the Dev experience much better when they're coupled together. 🤔